android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -111,21 +113,33 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
       // Only add the fragment once, when the activity is first created.
       if (savedInstanceState == null) {
         val fragment =
-          QuestionnaireFragment().apply {
+          BaseQuestionnaireFragment().apply {
             val parsedQuestionnaire = parser.encodeResourceToString(questionnaire)
             arguments =
               when {
                 clientIdentifier == null ->
                   bundleOf(Pair(BUNDLE_KEY_QUESTIONNAIRE, parsedQuestionnaire))
                 clientIdentifier != null -> {
-                  val parsedQuestionnaireResponse =
-                    parser.encodeResourceToString(
-                      questionnaireViewModel.generateQuestionnaireResponse(questionnaire!!, intent)
+
+                  try {
+                    fhirEngine.load(Patient::class.java, clientIdentifier!!)
+                    val parsedQuestionnaireResponse =
+                      parser.encodeResourceToString(
+                        questionnaireViewModel.generateQuestionnaireResponse(
+                          questionnaire!!,
+                          intent
+                        )
+                      )
+                    bundleOf(
+                      Pair(BUNDLE_KEY_QUESTIONNAIRE, parser.encodeResourceToString(questionnaire)),
+                      Pair(BUNDLE_KEY_QUESTIONNAIRE_RESPONSE, parsedQuestionnaireResponse)
+                    )
+                  } catch (e: Exception) {
```

Comment:
can you catch something more specific here?

Reply:
done